### ISSUE TITLE
Create Fixed Point Signed Safe Library

### ISSUE THREAD

**Issue by aparnakr** [OpenZeppelin/openzeppelin-contracts#2265]
**🧐 Motivation**
Currently, there is no good fixed point math library out there and each project is rolling out their own library.

**📝 Details**
A signed (and unsigned) fixed point math library that allows for addition, subtraction, multiplication, division. In the future, a more advanced library that allows for exponents, ln, log, square root.

**Comment by nventuro** [comment]
Hello @aparnakr!

Fixed point math support was [part of our Q2 Roadmap](https://github.com/OpenZeppelin/openzeppelin-contracts/issues/2207), but development [halted](https://forum.openzeppelin.com/t/designing-fixed-point-math-in-openzeppelin-contracts/2499/26) once the Solidity team recently decided they'd implement it natively, following pretty much the same design patterns we had explored.

This was about a month ago, but I don't know if there have been any recent updates. Discussion was happening in https://github.com/ethereum/solidity/issues/409.

Once that feature is released, we'll likely want to build some libraries on top of it, perhaps including advanced operations like exponentiation and the like, but that depends on what the Solidity team delivers.
